src/inference/src/ie_core.cpp

```diff
@@ -494,13 +494,45 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             res = LoadNetworkFromCache(cacheManager, hash, plugin, parsed._config, context, loadedFromCache);
             if (!loadedFromCache) {
                 res = compile_model_impl(network, plugin, parsed._config, context, hash);
+            } else {
+                updateCNNInfo(res, network);
             }
         } else {
             res = compile_model_impl(network, plugin, parsed._config, context, {});
         }
         return res;
     }
 
+    static void updateCNNInfo(ie::SoExecutableNetworkInternal& exe_net, const ie::CNNNetwork& network) {
+        // Temporary workaround until all plugins support caching of original model inputs
+        OPENVINO_SUPPRESS_DEPRECATED_START
+        exe_net->setNetworkInputs(network.getInputsInfo());
+        exe_net->setNetworkOutputs(network.getOutputsInfo());
+        {
+            std::vector<std::shared_ptr<const ov::Node>> params;
+            const auto& orig_params = network.getFunction()->get_parameters();
+            std::transform(orig_params.begin(),
+                           orig_params.end(),
+                           std::back_inserter(params),
+                           [](const std::shared_ptr<op::v0::Parameter>& p) {
+                               return p;
```

Comment:
the same - copy param? otherwise, I can change layout in user's code

Reply:
Fixed